### DISCUSSION TITLE
Is it safe to use HttpParser or is it internal API?

### DISCUSSION THREAD

**Discussion by kiminuo** [dotnet/aspnetcore#28500] [Q&A] [answered]
Hi,

https://github.com/dotnet/aspnetcore/blob/master/src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs is marked as public but the namespace contains `internal` word.

Can I use the class in my project? Or the API may be removed?

Thanks

**Comment by gfoidl** [comment, marked as answer]
This is a so called "pubternal" type. 
It's access modifier is public (dictated by the usage), but inside an "internal"-namespace, so it should be considered as being an internal-type / implementation detail which doesn't follow the contracts of public APIs.

That means you can use it, but at your own risk.
There may be breaking changes (remember: it's "internal").

You could use the code (not the binary) in your project if you follow the [license](https://github.com/dotnet/aspnetcore/blob/master/LICENSE.txt), to guard yourself from the beforementioned points.
